nautobot/extras/forms/mixins.py

```diff
@@ -72,6 +72,15 @@ def __init__(self, *args, **kwargs):
                 self.fields[field_name] = cf.to_form_field(
                     set_initial=False, enforce_required=False, simple_json_filter=True
                 )
+            elif cf.type == "date":
+                # TODO 2.0: revisit this when we do our filterset streamlining/consolidation for the v2 work
+                self.fields[field_name] = cf.to_form_field(set_initial=False, enforce_required=False)
+                self.fields[f"{field_name}__gte"] = cf.to_form_field(
+                    label=f"{str(cf)} (on or after this date)", set_initial=False, enforce_required=False
+                )
+                self.fields[f"{field_name}__lte"] = cf.to_form_field(
+                    label=f"{str(cf)} (on or before this date)", set_initial=False, enforce_required=False
+                )
```

Comment:
We don't add extra filter form fields to the UI for most other non-default filters (`__nic`, `__nisw`, etc.) - do we need/want these specifically for date custom fields, or is this something that we should just leave out until we get to #1998?

Reply:
If we want them to be used in the filter forms they have to match. I would say let's just do it the sway until we get to #1998. And then do all the backwards-incompatinble things in v2.

Reply:
Upon revisiting this I agree with @glennmatthews and we should not add the form fields here since these are extended lookups and we haven't been doing it for the others.